### PR TITLE
fix: correct Tamil year (Vishvavasu) calculation

### PR DESCRIPTION
**Pull request by rdukb**
## Summary
- fix Tamil year cycle anchor used for Samvatsara name mapping
- fix pre-Mesha-Sankranti year selection logic
- add focused tests for known years and Feb/Apr 2026 boundary behavior

## Why
On dates like 2026-02-07 (before Mesha Sankranti), UI showed `Vijaya` when expected year is `Vishvavasu`.

## Validation
- `PYTHONPATH=. ./venv/bin/pytest tests/test_tamil_calendar.py -q`

### CODE REVIEW

File: backend/src/services/calculations/tamil_calendar.py [hunk 1]
```diff
@@ -50,17 +50,11 @@ def calculate_tamil_calendar(julian_day: float) -> dict:
...
-    # More accurate: check if before Mesha Sankranti
-    if rashi_index < 0 or (rashi_index == 0 and sun_long_sidereal < 5):
-        tamil_year_num = year - 1
-    else:
-        tamil_year_num = year
+    # Tamil year calculation:
+    # year changes at Mesha Sankranti (sidereal Sun entering Aries, 0°).
+    # Before that transition (roughly Jan-mid Apr), use previous Gregorian year.
+    year, _, _, _ = swe.revjul(julian_day)
+    tamil_year_num = year - 1 if sun_long_sidereal >= 270 else year
```
[the remark this answers is not in the document]
Comment (rdukb) [reply backend/src/services/calculations/tamil_calendar.py]: Addressed in follow-up PR #16 by tightening Jan-Apr Tamil year boundary logic and adding explicit month-aware transition handling around Mesha Sankranti.

File: backend/tests/test_tamil_calendar.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    assert get_tamil_year_name(2025) == "Vishvavasu"
+    assert get_tamil_year_name(2026) == "Parabhava"
+
+
+def test_calculate_tamil_calendar_pre_mesha_uses_previous_tamil_year():
+    # February 7, 2026 (before Mesha Sankranti) should still be the 2025 Tamil year.
+    jd = swe.julday(2026, 2, 7, 12.0)
+    result = calculate_tamil_calendar(jd)
+    assert result["tamil_year"] == "Vishvavasu"
+
```
[the remark this answers is not in the document]
Comment (rdukb) [reply backend/tests/test_tamil_calendar.py]: Addressed in follow-up PR #16 by adding a January regression test (`2026-01-05`) to prevent boundary regressions.
